amazon-location-route-planner/src/App.vue

```diff
@@ -0,0 +1,349 @@
+    await calculateRoute();
+  }
+);
+
+watch(
+  () => parameters.truck.height,
+  async (value) => {
+    if (value === "") {
+      parameters.truck.height = null;
+    }
+    await calculateRoute();
+  }
+);
+
+watch(
+  () => parameters.truck.width,
+  async (value) => {
+    if (value === "") {
+      parameters.truck.width = null;
+    }
+    await calculateRoute();
+  }
+);
+
+watch(
+  () => parameters.truck.length,
+  async (value) => {
+    if (value === "") {
+      parameters.truck.length = null;
+    }
+    await calculateRoute();
+  }
+);
+
+watch(
+  () => parameters.truck.weight,
+  async (value) => {
+    if (value === "") {
+      parameters.truck.weight = null;
+    }
+    await calculateRoute();
+  }
+);
+
+// Initialize app /////////////////////////////////////////////////////////////////////////////////
+const initializeApp = async () => {
+  // Refresh auth credentials
+  await refreshCredentials();
+
+  // Initialize a new map instance
+  map = new maplibregl.Map({
+    container: "map",
+    center: [-114.067375, 51.046333],
+    zoom: 16,
+    style: mapName,
+    hash: true,
+    transformRequest,
+  });
+
+  // Add navigation controls for the map
+  map.addControl(new maplibregl.NavigationControl(), "bottom-right");
+
+  map.on("load", () => {
+    // Add a layer for rendering departure, waypoint, and destination positions on the map
+    map.addLayer({
+      id: "positions",
+      type: "circle",
+      source: { type: "geojson", data: positions },
+      paint: {
+        "circle-radius": 5,
+        "circle-color": "#ffffff",
+        "circle-stroke-color": "#00b0ff",
+        "circle-stroke-width": 3,
+      },
+    });
+
+    // Add a layer for rendering position labels on the map
+    map.addLayer({
+      id: "positions-label",
+      type: "symbol",
+      source: { type: "geojson", data: positions },
+      layout: {
+        "text-field": ["get", "title"],
+        "text-variable-anchor": ["left"],
+        "text-radial-offset": 0.5,
+        "text-justify": "auto",
+        "text-font": ["Fira GO Regular"],
+      },
+    });
+
+    // Add a layer for rendering routes on the map
+    map.addLayer(
+      {
+        id: "route",
+        type: "line",
+        source: { type: "geojson", data: route },
+        layout: {
+          "line-join": "round",
+          "line-cap": "round",
+        },
+        paint: {
+          "line-color": "#00b0ff",
+          "line-width": 5,
+          "line-opacity": 1,
+        },
+      },
+      "positions"
+    );
+  });
+
+  // Add an event handler to capture departure, waypoint, and destination positions
+  let counter = 0;
+  map.on("click", async (e) => {
+    counter++;
+    const { lngLat } = e;
+    const p = point([lngLat.lng.toFixed(5), lngLat.lat.toFixed(5)], {
+      title: `Point ${counter}`,
+    });
+    positions.features.push(p);
+  });
+};
+
+// Calculate route
+const calculateRoute = async () => {
+  const client = new LocationClient({
```

Comment:
I would recommend creating the client outside of the scope of this function so that it's instantiated only once and not every time the function is run

Reply:
great feedback. fixed it in [aee5817](https://github.com/aws-samples/amazon-location-samples/pull/247/commits/aee5817296be0322c99adba4deb933c638e358b2).